components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2186,6 +2186,23 @@ where
         if is_leader {
             self.fsm.peer.approximate_size = estimated_size;
             self.fsm.peer.approximate_keys = estimated_keys;
+
+            if let Some(estimated_size) = self.fsm.peer.approximate_size {
```

Comment:
If some followers do not apply the `SplitRequest`, this split check task may fail because of the leader can not propose a `SplitRequest` in this term. Shall we change all `ApproximateSize` task to `SplitCheckTask` ? Because all peers will check split when they win an election and become leader.

Reply:
this is called on the derived region, not on the newly split regions.